src/bp/core/services/converse.ts

```diff
@@ -55,25 +58,42 @@ export class ConverseService {
     })
   }
 
-  public async sendMessage(botId: string, userId: string, payload): Promise<any> {
-    if (!payload.text || !_.isString(payload.text) || payload.text.length > 360) {
-      throw new InvalidParameterError('Text must be a valid string of less than 360 chars')
+  public async sendMessage(botId: string, userId: string, conversationId: string, payload): Promise<any> {
+    if (!payload.text || !_.isString(payload.text)) {
+      throw new InvalidParameterError('Text must be a valid string')
     }
 
     await this.userRepository.getOrCreate('api', userId)
 
+    if (!conversationId) {
+      conversationId = await this.convoRepository.getOrCreateConversation(botId, userId, {
+        lifetime: this.conversationLifetime
+      })
+    }
+
     const incomingEvent = Event({
       type: 'text',
       channel: 'api',
       direction: 'incoming',
       payload,
       target: userId,
+      threadId: conversationId,
```

Comment:
We should try top use same vocabulary all around so either user `thread` or `conversation` but not both. I know we use thread in events and that might be misleading. Maybe we would implement a migration from conversation_web directly to thread and rename every services / repository ?

... food for thought and discussion ...

Reply:
Yep... i'd prefer threadId or convoId, conversation is too long for nothing